File: C++/1197.cpp
#include <algorithm>
#include <array>
#include <iostream>
#include <string>

struct Point {
  int x;
  int y;
};

struct Desk {
  static constexpr int SIZE = 8;
  static constexpr int KNIGHT_MOVES = 8;

  static bool isValid(int x, int y) {
    return x >= 0 && x < SIZE && y >= 0 && y < SIZE;
  }

  static constexpr std::array<Point, KNIGHT_MOVES> knightMoves = {
      {{-2, 1}, {-1, 2}, {1, 2}, {2, 1}, {2, -1}, {1, -2}, {-1, -2}, {-2, -1}}};
};

void runTest() {
  std::string cell;
  std::cin >> cell;

  Point position{cell[0] - 'a', cell[1] - '1'};
  auto isValidMove = [position](const Point& move) {
    return Desk::isValid(position.x + move.x, position.y + move.y);
  };

  int cnt = std::count_if(Desk::knightMoves.begin(), Desk::knightMoves.end(),
                          isValidMove);
  std::cout << cnt << '\n';
}

int main() {
  int tests;
  std::cin >> tests;
  for (int i = 0; i < tests; ++i) {
    runTest();
  }
  return 0;
}

File: Python/1197.py
SIZE = 8
DX = [-2, -1, 1, 2, 2, 1, -1, -2]
DY = [1, 2, 2, 1, -1, -2, -2, -1]

def isValid(x, y):
    return 0 <= x < SIZE and 0 <= y < SIZE

def runTest():
    cell = input()
    x = ord(cell[0]) - ord('a')
    y = int(cell[1]) - 1
    cnt = 0
    for dx, dy in zip(DX, DY):
        cnt += int(isValid(x + dx, y + dy))
    print(cnt)

tests = int(input())
for _ in range(tests):
    runTest()
